types/es6-shim/index.d.ts

```diff
@@ -556,7 +555,7 @@ interface Map<K, V> {
     clear(): void;
     delete(key: K): boolean;
     forEach(callbackfn: (value: V, index: K, map: Map<K, V>) => void, thisArg?: any): void;
-    get(key: K): V;
+    get(key: K): V | undefined;
     has(key: K): boolean;
     set(key: K, value?: V): Map<K, V>;
```

Comment:
Either `set` should not have an optional `value`, or the callback to `forEach` should type value as `V | undefined`. I would err on the side of the lib definition in TypeScript and remove the optionality of `value` in `set`.